### PR TITLE
Support /dev/lxd in VMs

### PR DESCRIPTION
**Pull request by monstermunchkin**

### CODE REVIEW

File: lxd-agent/events.go [hunk 1]
```diff
@@ -32,7 +35,7 @@ func (r *eventsServe) String() string {
 func eventsSocket(d *Daemon, r *http.Request, w http.ResponseWriter) error {
 	typeStr := r.FormValue("type")
 	if typeStr == "" {
-		typeStr = "logging,operation,lifecycle"
+		typeStr = "logging,operation,lifecycle,config"
```
Comment (stgraber) [inline lxd-agent/events.go:38]: Why is that needed?
That feels wrong since we'll never hit eventsGet for a `config` event type

Comment (monstermunchkin) [reply lxd-agent/events.go:38]: When listening on `/dev/lxd/sock` for a config change, we need this here. Otherwise the user won't get notified.

Comment (stgraber) [reply lxd-agent/events.go:38]: Oh, I see, we're abusing eventsGet from within devlxd :)

**Comment by stgraber** [comment]
@monstermunchkin so one more thing we need to do I think.

Let's only generate `instance-data` if `security.devlxd` is true and have `lxd-agent` only start the `devlxd` listener if an `instance-data` file is present.

File: lxd-agent/events.go (hunk 1 above)
Comment (stgraber) [reply lxd-agent/events.go:38]: @monstermunchkin can you add a comment in there so we remember why this is done that way?

**Comment by monstermunchkin** [comment]
Fixed
